src/sknnr_spatial/image/_base.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sized
+from functools import singledispatch
+from types import ModuleType
+from typing import TYPE_CHECKING
+
+import numpy as np
+from numpy.typing import NDArray
+
+if TYPE_CHECKING:
+    import xarray as xr
+    from sklearn.base import BaseEstimator
+    from sklearn.neighbors import KNeighborsRegressor
+
+    from ..types import ImageType
+
+
+class ImagePreprocessor(ABC):
+    """
+    A pre-processor for multi-channel image data in a machine learning workflow.
+
+    This class handles flattening an image from 3D pixel space (e.g. y, x, channel)
```

Comment:
I'm being inconsistent with wording between `band` and `channel` throughout... Do you have a preference for what we should use here? Technically there's nothing that makes `sknnr-spatial` specific to geospatial rasters, so maybe band isn't general enough?

Reply:
You're right, neither one really makes sense outside of optical data! I think I'll stick with the `rasterio` convention and use `band` for now just for lack of a better option, but calling that out in the docs is a good idea.